### PR TITLE
[chore] Update opentelemetry to v1.20.0

### PR DESCRIPTION
**Pull request by tmshort**
Update to match operator-framework/api

**Description of the change:**

**Motivation for the change:**

**Architectural changes:**

**Testing remarks:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Bug fixes are accompanied by regression test(s)
- [ ] e2e tests and flake fixes are accompanied evidence of flake testing, e.g. executing the test 100(0) times
- [ ] tech debt/todo is accompanied by issue link(s) in comments in the surrounding code
- [ ] Tests are comprehensible, e.g. Ginkgo DSL is being used appropriately
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky and have an issue
- [ ] Code is properly formatted

### CODE REVIEW

**Review by m1kola — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -11,34 +11,34 @@ require (
...
-	github.com/onsi/ginkgo/v2 v2.9.5
-	github.com/onsi/gomega v1.27.7
+	github.com/mitchellh/mapstructure v1.5.0
+	github.com/onsi/ginkgo/v2 v2.12.0
+	github.com/onsi/gomega v1.27.10
 	github.com/openshift/api v3.9.0+incompatible
 	github.com/openshift/client-go v0.0.0-20220525160904-9e1acff93e4a
 	github.com/operator-framework/api v0.19.0
-	github.com/operator-framework/operator-registry v1.32.0
+	github.com/operator-framework/operator-registry v1.32.1-0.20231119153434-5e23ef594a41
```
Comment (grokspawn) [inline go.mod:28]: Any reason we need this pre-release tag?  v1.32.0 has opentelemetry at v1.20 already.

Comment (grokspawn) [reply go.mod:28]: /hold to check out the number here

Comment (tmshort) [reply go.mod:28]: opentelemetry was updated in the last operator-registry commit which is not included in v1.32.0

Comment (tmshort) [reply go.mod:28]: Are you sure? The tip commit of operator-registry updates opentelemetry:
https://github.com/operator-framework/operator-registry/commit/5e23ef594a41e6c8ce843d48b22715319c684dff

**Comment by tmshort** [comment]
@grokspawn updated to use the tag for operator registry

**Review by grokspawn — APPROVED** [review]

**Pull request merged**